### PR TITLE
Refactor drp grid

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Refactored date range picker calendar grid. See updates description in comments.

### How has this been tested?

Manually tested space selection of calendar days.

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [x] _No._

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/calendar/grid/index.tsx [hunk 1]
```diff
@@ -53,6 +53,7 @@ export default function Grid({
     }
 
     switch (event.keyCode) {
+      case KeyCode.space:
```
Comment (pan-kot) [inline src/calendar/grid/index.tsx:56]: Added this for consistency with DRP where the date can be selected with both SPACE and ENTER.

File: src/date-range-picker/calendar/grids/grid.tsx [hunk 2]
```diff
@@ -2,14 +2,26 @@
 // SPDX-License-Identifier: Apache-2.0
 import React, { useMemo } from 'react';
 import styles from '../../styles.css.js';
-import GridDay from './day';
```
Comment (pan-kot) [inline src/date-range-picker/calendar/grids/grid.tsx:5]: Removed the grid day file for the layout to be only defined in one file avoiding the extra interfaces. That is consistent with Calendar grid implementation.

File: src/date-range-picker/calendar/grids/index.tsx [hunk 3]
```diff
@@ -114,6 +114,7 @@ export const Grids = ({
     }
 
     switch (e.keyCode) {
+      case KeyCode.space:
```
Comment (pan-kot) [inline src/date-range-picker/calendar/grids/index.tsx:117]: This was previously handled with a separate onKeyDown handler attached to every day button.

**Review by karmanya79 — APPROVED** [review]

File: src/calendar/grid/index.tsx (hunk 1 above)
Comment (just-boris) [reply src/calendar/grid/index.tsx:56]: If the logic between two components is the same, why is it not a `useKeyboardNavigation` hook or something similar?

Comment (pan-kot) [reply src/calendar/grid/index.tsx:56]: I like the idea and will check if that can be done as a follow-up 👍 

Hopefully we can reuse more code between calendar and drp calendar eventually, currently the implementations have a lot of inconsistencies which can probably be resolved.

**Review by YueyingLu — APPROVED** [review]

**Pull request merged**